File: frontend/dto/baseAlbumSongResponse.ts
import { z } from 'zod';

export const BaseAlbumSongResponseSchema = z.object({
    provider: z.string(),
    publicId: z.string(),
    name: z.string(),
    artists: z.any(),
    audioSrc: z.any(),
    downloaded: z.boolean(),
    internalImageUrl: z.string(),
});

export type BaseAlbumSongResponse = z.infer<typeof BaseAlbumSongResponseSchema>;

File: backend/core/responses/baseAlbumSongResponse.py
from typing import Sequence
from pydantic import BaseModel

from backend.core.responses.baseArtistResponse import BaseArtistResponse


class BaseAlbumSongResponse(BaseModel):
    provider: str
    publicId: str
    name: str
    artists: Sequence[BaseArtistResponse]
    audioSrc: str | None
    downloaded: bool
    internalImageUrl: str
    duration: int
    discNumber: int
    trackNumber: int
